language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -55,14 +50,46 @@ optionsParserInfo = info (optionsParser <**> helper)
 main :: IO ()
 main = do
     opts@Options{..} <- execParser optionsParserInfo
-    dar <- B.readFile optInputDar
-    let archive = Zip.toArchive $ BSL.fromStrict dar
-    dalfs <- either fail pure $ DAR.readDalfs archive
-    DAR.DalfManifest{packageName, ..} <- either fail pure $ DAR.readDalfManifest archive
-    packageName <- pure $ optMainPackageName <|> packageName
-    forM_ ((DAR.mainDalf dalfs, packageName) : map (, Nothing) (DAR.dalfs dalfs)) $ \(dalf, mbPkgName) -> do
-        (pkgId, pkg) <- either (fail . show)  pure $ Archive.decodeArchive Archive.DecodeAsMain (BSL.toStrict dalf)
-        daml2ts opts pkgId pkg mbPkgName
+    foldM_ (processDar opts) Map.empty optInputDars
+      where
+        -- Generate the ts for a single DAR. 'processed' is a map of
+        -- package ids of processed DALFs (the same package can appear
+        -- in multiple DARS - avoid regenerating them where possible).
+        processDar :: Options -> Map.Map PackageId [String] -> FilePath -> IO (Map.Map PackageId [String])
```

Comment:
It would be good to have a description of what it means for a `PackageId` to appear as a key of the map. And what it means for a `String` to be contained in the list of a specific `PackageId`.

Reply:
Ok.